react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,49 @@
+import { Images } from './Images';
+import { NextButton } from './NextButton';
+import { PrevButton } from './PrevButton';
+import './Indicators.css';
+import { Indicators } from './Indicators';
+import { useEffect, useState } from 'react';
+
+export type Image = {
+  src: string;
+  alt: string;
+};
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentImage, setCurrentImage] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setCurrentImage((currentImage + 1) % images.length);
+    }, 3000);
+    return () => clearTimeout(timerId);
+  });
```

Comment:
You need to include a dependency array! Always include a dependency array when you call `useEffect`.